packages/namers/default/src/DefaultNamer.js

```diff
@@ -37,7 +38,7 @@ export default (new Namer({
 
     if (
       bundle.id === mainBundle.id &&
-      bundle.needsStableName &&
+      isEntry &&
```

Comment:
Shouldn't we keep `bundle.needsStableName &&`?

Reply:
Good question. I assumed that the target's configured distEntry should always win?

Reply:
This is correct if we say that `needsStableName == true` implies that there is no hash in the bundle name, and not necessarily the other way around.